bug: build script occurs error when use make undeps command

### Current Behavior

While building APISIX from source on Ubuntu 22.04, I encountered errors with the make undeps commands.

```
[ info ] undeps -> [ Start ]
luarocks purge --tree=deps
/bin/bash: /usr/local/bin/luarocks: /usr/local/openresty/luajit/bin/luajit: bad interpreter: No such file or directory
make: *** [Makefile:147: undeps] Error 126
```

**Cause**: The make undeps command attempts to run uninstall-runtime first, which removes the /usr/local/openresty directory, causing luajit to be unavailable for luarocks purge --tree=deps.

```
// Makefile
### undeps : Uninstalling dependencies
.PHONY: undeps
undeps: uninstall-runtime
	@$(call func_echo_status, "$@ -> [ Start ]")
	$(ENV_LUAROCKS) purge --tree=deps
	@$(call func_echo_success_status, "$@ -> [ Done ]")

.PHONY: uninstall-runtime
uninstall-runtime:
	./utils/install-dependencies.sh uninstall
	rm -rf /usr/local/openresty
	rm -f $(ENV_INST_BINDIR)/openresty
```

**Solution**: Modify the process to run luarocks purge --tree=deps before removing the /usr/local/openresty directory to ensure that luajit is available when needed.

### Expected Behavior

No errors when build apisix from source at ubuntu22.04

### Error Logs

_No response_

### Steps to Reproduce

1. clone apisix
2. apt install libyaml-dev(manully install libyaml-dev for now, see https://github.com/apache/apisix/issues/11290)
3. run make deps && make installl
5. run make uninstall && make undeps

### Environment

- Operating system (run `uname -a`): Linux ubuntu 5.15.0-107-generic # 117-Ubuntu SMP Mon Apr 29 14:37:09 UTC 2024 aarch64 aarch64 aarch64 GNU/Linux

--- a/Makefile
+++ b/Makefile
@@ -142,7 +142,11 @@ deps: install-runtime
 
 ### undeps : Uninstalling dependencies
 .PHONY: undeps
-undeps: uninstall-runtime
+undeps: uninstall-rocks uninstall-runtime
+
+
+.PHONY: uninstall-rocks
+uninstall-rocks:
 	@$(call func_echo_status, "$@ -> [ Start ]")
 	$(ENV_LUAROCKS) purge --tree=deps
 	@$(call func_echo_success_status, "$@ -> [ Done ]")

